worktrolly-ui/src/app/services/auth.service.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable } from '@angular/core';
+import firebase from 'firebase/app';
+import { AngularFireAuth } from '@angular/fire/auth';
+import { AngularFirestore, AngularFirestoreDocument } from '@angular/fire/firestore';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { User } from "../Interface/UserInterface";
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+
+  loginForm: FormGroup = new FormGroup({
+    email: new FormControl("", [Validators.required, Validators.email]),
+    password: new FormControl("", [Validators.required, Validators.minLength(6)]),
+  });
+  signupForm: FormGroup = new FormGroup({
+    username: new FormControl("", [Validators.required, Validators.minLength(3)]),
+    email: new FormControl("", [Validators.required, Validators.email]),
+    password: new FormControl("", [Validators.required, Validators.minLength(6)]),
+  });
+  resetLoginForm() {
+    this.loginForm.setValue({
+      email: "",
+      password: ""
+    })
+  }
+  resetSignupForm() {
+    this.signupForm.setValue({
+      username: "",
+      email: "",
+      password: ""
+    })
+  }
+
+  constructor(public afauth: AngularFireAuth, public afs: AngularFirestore) { }
+
+  private createUserData(user: User) {
+    // Sets user data to firestore on login
+    const userRef: AngularFirestoreDocument<User> = this.afs.doc(
+      `Users/${user.uid}`
+    );
```

Comment:
Writing of user data into db can be done as api, making the db less exposed and secured, and also the process will become fast, and will be handled parallely